Rancher 2.5 (Single Install) not starting after nf_conntrack_max value adjustment

**What kind of request is this (question/bug/enhancement/feature request):** bug


**Steps to reproduce (least amount of steps as possible):**

In this scenario, the Docker Host where the Rancher container is started is a LXC container.

```
root@ubuntu4:~# sysctl -a|grep conntrack_max
net.netfilter.nf_conntrack_max = 524288

root@ubuntu4:~# docker run -d --name rancher --restart=unless-stopped -p 80:80 -p 443:443 --privileged rancher/rancher:v2.5.8
```

Installation and setup works fine.

Now the nf_conntrack_max value is set to another value on the LXC host:

```
root@host ~ # sysctl -w net.netfilter.nf_conntrack_max=525000
net.netfilter.nf_conntrack_max = 525000
```

Inside the LXC container, the nested Rancher container is restarted:

```
root@ubuntu4:~# docker restart rancher

root@ubuntu4:~# docker logs --follow rancher
[...]
I0624 07:38:23.051838      54 node.go:136] Successfully retrieved node IP: 172.17.0.2
I0624 07:38:23.051867      54 server_others.go:143] kube-proxy node IP is an IPv4 address (172.17.0.2), assume IPv4 operation
I0624 07:38:23.052887      54 server_others.go:186] Using iptables Proxier.
I0624 07:38:23.053288      54 server.go:650] Version: v1.19.8+k3s1
I0624 07:38:23.053960      54 conntrack.go:103] Set sysctl 'net/netfilter/nf_conntrack_max' to 524288
F0624 07:38:23.053999      54 server.go:495] open /proc/sys/net/netfilter/nf_conntrack_max: permission denied
2021/06/24 07:38:23 [FATAL] k3s exited with: exit status 1
```


**Result:**
Start of Rancher container fails. At least if the current value is same or higher than the remembered value.


**Expected Result:**
Start of Rancher container succeeds. 


**Other details that may be helpful:**

The problem seems to be in the func `realConntracker` in k3s/conntrack.go:
https://github.com/k3s-io/k3s/blob/master/vendor/k8s.io/kubernetes/cmd/kube-proxy/app/conntrack.go#L98

This function checks the current sysctl value and compares it to a (somewhere internally retained?) internal value:

```
func (realConntracker) setIntSysCtl(name string, value int) error {
	entry := "net/netfilter/" + name

	sys := sysctl.New()
	if val, _ := sys.GetSysctl(entry); val != value {
		klog.Infof("Set sysctl '%v' to %v", entry, value)
		if err := sys.SetSysctl(entry, value); err != nil {
			return err
		}
	}
	return nil
}
```

If the current system value doesn't match the internal value, sys.SetSysctl is launched to set the previously known value. Yet this fails in a LXC container as the value is defined at Host level:

```
root@ubuntu4:~# sysctl -w net/netfilter/nf_conntrack_max=524288
root@ubuntu4:~# sysctl -a|grep conntrack_max
net.netfilter.nf_conntrack_max = 525000
```

Probably related: https://github.com/kubernetes-sigs/kind/pull/2241 


**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.5.8
- Installation option (single install/HA): Single Install

- Docker version (use `docker version`):


```
root@ubuntu4:~# docker version
Client:
 Version:           20.10.2
 API version:       1.41
 Go version:        go1.13.8
 Git commit:        20.10.2-0ubuntu1~20.04.2
 Built:             Tue Mar 30 21:24:57 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.2
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.8
  Git commit:       20.10.2-0ubuntu1~20.04.2
  Built:            Mon Mar 29 19:10:09 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.4-0ubuntu1~20.04.2
  GitCommit:        
 runc:
  Version:          1.0.0~rc95-0ubuntu1~20.04.1
  GitCommit:        
 docker-init:
  Version:          0.19.0
  GitCommit:        

```

This also affects/breaks deploying a Rancher managed Kubernetes cluster to nodes. The Kubernetes deployment is stuck on kube-proxy deployment. kube-proxy tries to set a specific value as nf_conntrack_max but fails:

```
I0723 07:30:08.943595   12314 node.go:172] Successfully retrieved node IP: 172.17.0.1
I0723 07:30:08.943627   12314 server_others.go:142] kube-proxy node IP is an IPv4 address (172.17.0.1), assume IPv4 operation
W0723 07:30:08.944553   12314 server_others.go:584] Unknown proxy mode "", assuming iptables proxy
I0723 07:30:08.944629   12314 server_others.go:182] DetectLocalMode: 'ClusterCIDR'
I0723 07:30:08.944652   12314 server_others.go:185] Using iptables Proxier.
I0723 07:30:08.944760   12314 proxier.go:287] iptables(IPv4) masquerade mark: 0x00004000
I0723 07:30:08.944857   12314 proxier.go:334] iptables(IPv4) sync params: minSyncPeriod=1s, syncPeriod=30s, burstSyncs=2
I0723 07:30:08.944977   12314 proxier.go:346] iptables(IPv4) supports --random-fully
I0723 07:30:08.945162   12314 server.go:650] Version: v1.20.8
I0723 07:30:08.945589   12314 conntrack.go:100] Set sysctl 'net/netfilter/nf_conntrack_max' to 131072
F0723 07:30:08.945634   12314 server.go:495] open /proc/sys/net/netfilter/nf_conntrack_max: permission denied
```

kube-proxy is then restarted in a loop, trying to fix it.

On Rancher server side, the following error is logged:

```
2021/07/23 07:40:55 [ERROR] Failed to upgrade worker components on NotReady hosts, error: [Failed to verify healthcheck: Failed to check http://localhost:10256/healthz for service [kube-proxy] on host [192.168.15.161]: Get "http://localhost:10256/healthz": dial tcp 127.0.0.1:10256: connect: connection refused, log: F0723 07:40:32.920751   19619 server.go:495] open /proc/sys/net/netfilter/nf_conntrack_max: permission denied]
```

This breaks Rancher Kubernetes deployments in LXC and LXD environments. 

My Kubernetes PR is still open unfortunately.

--- a/cmd/kube-proxy/app/conntrack.go
+++ b/cmd/kube-proxy/app/conntrack.go
@@ -96,7 +96,7 @@ func (realConntracker) setIntSysCtl(name string, value int) error {
 	entry := "net/netfilter/" + name
 
 	sys := sysctl.New()
-	if val, _ := sys.GetSysctl(entry); val != value {
+	if val, _ := sys.GetSysctl(entry); val != value && val < value {
 		klog.InfoS("Set sysctl", "entry", entry, "value", value)
 		if err := sys.SetSysctl(entry, value); err != nil {
 			return err

